apps/nextjs/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -2,47 +2,47 @@ import type { NextRequest } from "next/server";
 import { cookies } from "next/headers";
 import { NextResponse } from "next/server";
 
-import { GET as _GET, POST } from "@acme/auth";
+import { GET as DEFAULT_GET, POST } from "@acme/auth";
 
 export const runtime = "edge";
 
-const expoRedirectCookieName = "__acme-expo-redirect-state";
-const setCookieMatchPattern = /next-auth\.session-token=([^;]+)/;
+const EXPO_COOKIE_NAME = "__acme-expo-redirect-state";
+const AUTH_COOKIE_PATTERN = /authjs\.session-token=([^;]+)/;
 
 export const GET = async (
   req: NextRequest,
   props: { params: { nextauth: string[] } },
 ) => {
   const nextauthAction = props.params.nextauth[0];
   const isExpoSignIn = req.nextUrl.searchParams.get("expo-redirect");
-  const isExpoCallback = cookies().get(expoRedirectCookieName);
+  const isExpoCallback = cookies().get(EXPO_COOKIE_NAME);
 
   if (nextauthAction === "signin" && !!isExpoSignIn) {
     // set a cookie we can read in the callback
     // to know to send the user back to expo
     cookies().set({
-      name: expoRedirectCookieName,
+      name: EXPO_COOKIE_NAME,
       value: isExpoSignIn,
       maxAge: 60 * 10, // 10 min
       path: "/",
     });
   }
 
   if (nextauthAction === "callback" && !!isExpoCallback) {
-    cookies().delete(expoRedirectCookieName);
+    cookies().delete(EXPO_COOKIE_NAME);
 
-    const authResponse = await _GET(req);
+    const authResponse = await DEFAULT_GET(req);
     const setCookie = authResponse.headers.getSetCookie()[0];
```

Comment:
I was trying to use this solution with Google and it didn't work, after some debugging it turned out this cookie is actually at index 1, so I think this part needs to be a find or something